File: warehouse_management/warehouse_management/report/ledger_report/ledger_report.py
# Copyright (c) 2022, Sabu Siyad and contributors
# For license information, please see license.txt

import frappe

def execute(filters=None):
	columns = [
		{
			"fieldname": "modified",
			"fieldtype": "Datetime",
			"label": "Date",
		},
		{
			"fieldname": "item",
			"fieldtype": "Link",
			"label": "Item",
			"options": "Item",
		},
		{
			"fieldname": "warehouse",
			"fieldtype": "Link",
			"label": "Warehouse",
			"options": "Warehouse",
		},
		{
			"fieldname": "quantity_change",
			"fieldtype": "Int",
			"label": "Change in Quantity",
			"indicator": "Green"
		},
		{
			"fieldname": "value_change",
			"fieldtype": "Currency",
			"label": "Change in Value",
		},
	]

	data = frappe.db.get_list("Warehouse Ledger", filters=filters, fields="*")

	return columns, data

File: warehouse_management/warehouse_management/report/ledger_report/ledger_report.js
// Copyright (c) 2022, Sabu Siyad and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Ledger Report"] = {
	"formatter": function(value, row, column, data, default_formatter) {
		value = default_formatter(value, row, column, data);

		if (column.id == "value_change" && data) {
			color = data.value_change < 0 ? "red" : "green";
			value = `<span style='color:${color};'>${value}</span>`;
		}

		return value;
	},
};
